Feature: Implement pursing function

CeNTePeDe opened the pull request.

CeNTePeDe committed:
created:
- Dockerfile
- docker-compose.yml
- pipenv for virtualenv

CeNTePeDe committed:
added description to model

CeNTePeDe committed:
added MongoClient

CeNTePeDe committed:
added aiohttp, lxml
added function:
- get_page_data
- gather_data
added constant_variables with url

slavadan commented on main.py:
```diff
@@ -1,5 +1,10 @@
+)
+
 
+@app.get("/")
```
убрат

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,66 @@
+    async with aiohttp.ClientSession() as session:
+        response = await session.get(url, headers=headers)
+        soup = BeautifulSoup(await response.text(), "lxml")
+        number_of_products = int(soup.find("span", class_="d-catalog-header__product-counter").text.split(' ')[0])
```
' ' - убрать, он по стандарту пробелы удаляет

slavadan commented on models/product_model.py:
```diff
@@ -0,0 +1,21 @@
+    picture_link: HttpUrl
+    price: float = Field()
+    product_detail_link: HttpUrl
+    description: dict = Field()
```
dict

slavadan replied:
descrtiption -> characteristics или как там

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,66 @@
+from bs4 import BeautifulSoup
+import asyncio
+import aiohttp
+import requests
```
импорты отсортируй

slavadan commented on config/product_db.py:
```diff
@@ -0,0 +1,7 @@
+
+db = client["product_db"]
+
+collection = db["collection"]
```
отступ

slavadan replied:
Сделать 3 типа настроек

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,66 @@
+headers = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"}
+
+product_data = []
```
перенести в функцию, сделать тайпхинт на возвращение листа

CeNTePeDe committed:
added logging
created get_data_for_each_product
removed exceptions

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+            if picture_link is None:
+                picture_link = url
+            else:
+                picture_link = MAIN_PAGE + picture_link.get("src")
```
if picture: picture_link = MAIN_PAGE + picture_link.get("src")

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+            if price is None:
+                price = item.find("span", class_="x-product-card-description__price-new").text
+            else:
+                price = price.text
```
price = item.find("span", class_="x-product-card-description__price-single")
if price is None:
    price = item.find("span", class_="x-product-card-description__price-new")

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+            product_data.append({
+                "name": name,
+                "picture": picture_link,
+                "price": price,
```
price.text

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+                "link_to_product": scr_link,
+            })
+        product_data = get_data_for_each_product(product_data=product_data)
+        logger.debug(f"Parsing date is finished successfully on the page={page}")
```
info

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+        if description is None:
+            description = "No Content"
+        else:
+            description = description.text.strip("\n").strip()
```
description = html_item_product.find("div", class_="x-premium-product-description__text")
description = description.text.strip("\n").strip() if description else "No Content"

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+            characteristic_product_item.update({item_title.text.strip(): item_value.text.strip()})
+        product_item["characteristic"] = characteristic_product_item
+        product_item["description"] = description
+    logger.debug("Parsing date for each product is finished successfully")
```
info

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,96 @@
+            tasks.append(task)
+
+        await asyncio.gather(*tasks)
+    logger.debug("gather_data is finished successfully")
```
info

slavadan commented on core/constant_variables.py:
```diff
@@ -0,0 +1,9 @@
+
+
+HEADERS = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"}
```
отступ в конце файла

slavadan commented on docker-compose.yml:
```diff
@@ -0,0 +1,26 @@
+      - "27017:27017"
+
+volumes:
+  data:
```
отступ

CeNTePeDe committed:
removed log configuration from parse_product
created asyncio get_data_for_each_product
created files
 - dev.py
 - stage.py
 - prod.py

slavadan commented on config/product_db.py:
```diff
@@ -0,0 +1,10 @@
+
+db = client[settings.MONGODB_DB]
+
+collection = db[settings.PRODUCT_COLLECTION]
```
отступ в конце файла

slavadan commented on config/prod.py:
```diff
@@ -0,0 +1,52 @@
+        "handlers": ["console"],
+        "level": "DEBUG",
+    },
+}
```
Ну обсуждали же, что конфиги должны находиться в одном классе. + отступ в конце файла

slavadan commented on config/stage.py:
```diff
@@ -0,0 +1,52 @@
+        "handlers": ["console"],
+        "level": "DEBUG",
+    },
+}
```
Лютый дубликат

slavadan commented on config/stage.py:
```diff
@@ -0,0 +1,52 @@
+settings = Settings()
+
+LOGGING_CONFIG = {
+    "version": 1,
```
Весь этот файл лютый дубликат. Ты же вкурсе, что можно наследоваться?

slavadan commented on core/constant_variables.py:
```diff
@@ -0,0 +1,5 @@
+MAIN_PAGE = "https://www.lamoda.by"
+
+MAIN_PAGE_MEN = "https://www.lamoda.by/men-home/"
+MAIN_PAGE_WOMAN = "https://www.lamoda.by/women-home/"
+MAIN_PAGE_KIDS = "https://www.lamoda.by/kids-home/"
```
отступ вконце файла

slavadan commented on docker-compose.yml:
```diff
@@ -0,0 +1,27 @@
+      - "27017:27017"
+
+volumes:
+  mongodb:
```
Отступ вконец файла

slavadan commented on main.py:
```diff
@@ -1,5 +1,15 @@
 
+logging.config.dictConfig(LOGGING_CONFIG)
+
+app = FastAPI(title=settings.PROJECT_NAME)
```
инициализация клиентов, бд и других подобны вещей можно вынести в одни файл.

slavadan commented on models/product_model.py:
```diff
@@ -0,0 +1,24 @@
+
+
+class UrlSchema(BaseModel):
+    url: HttpUrl
```
отступ!

slavadan commented on parsing_function/parse_product.py:
```diff
@@ -0,0 +1,105 @@
+            for item_title, item_value in connect_value:
+                characteristic.update(
+                    {item_title.text.strip(): item_value.text.strip()}
+                )
```
после цикла делается отступ

slavadan commented on config/dev.py:
```diff
@@ -0,0 +1,52 @@
+        env_file_encoding = "utf-8"
+
+
+settings = Settings()
```
вынести. И сделать так чтобы сетинги подтягивались в зависимости от окружения. А то толку 0 от двух типов настроек, если у тебя всё завязано под dev конфиг

CeNTePeDe committed:
added to settings:
- set_environment
inherit SettingsStage, SettingsProd from StageDev

slavadan commented on main.py:
```diff
@@ -1,5 +1,12 @@
 
-app = FastAPI()
+from routers import product
+from settings import environment
```
отступы

slavadan commented on models/product_model.py:
```diff
@@ -0,0 +1,17 @@
+    characteristic: Optional[dict]
+
+
+class UrlSchema(BaseModel):
+    url: AnyUrl
```
чо это?

slavadan commented on settings.py:
```diff
@@ -0,0 +1,23 @@
+    if environment == "stage":
+        return config.stage
+    else:
+        raise ValueError("Invalid environment")
```
неправильно, подумой еще

slavadan commented on settings.py:
```diff
@@ -0,0 +1,23 @@
+        raise ValueError("Invalid environment")
+
+
+env = os.environ.setdefault("ENV", "dev")
```
ябы назвал APPLICATION_CONFIG

slavadan commented on settings.py:
```diff
@@ -0,0 +1,23 @@
+        return config.dev
+    if environment == "prod":
+        return config.prod
+    if environment == "stage":
```
ну вообще settings и конфиг одно и тоже. Я вроде говорил вчера, что инициалищация происходит в основном файле

slavadan commented on config/prod.py:
```diff
@@ -0,0 +1,9 @@
+    pass
+
+
+settings: SettingsProd = SettingsProd()
```
инициалищация 3 типов настроек бессмысленная трате памяти тк в одном окружении будет использоваться только 1 класс с настройками. инициализируй settings в основном файле.

CeNTePeDe committed:
override init_settings function
